From 5a4a4ebb462d8033194194fa3950555ce6e1e957 Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Fri, 9 Oct 2015 17:25:17 +0800
Subject: [PATCH 0534/5242] MLK-11679: IPU: Fix ldb0 can not work issue

commit  862135d5dba54a9cfcf704bbb1626fa7505cb11b from
https://source.codeaurora.org/external/imx/linux-imx.git

LVDS0 can not work on imx6q auto and SDB board,
it is caused by ldb0 clock setting is missed in ipu driver.

Signed-off-by: Sandor Yu <R01008@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/ipu3/ipu_disp.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mxc/ipu3/ipu_disp.c b/drivers/mxc/ipu3/ipu_disp.c
index a983ba7..fa73bfa 100644
--- a/drivers/mxc/ipu3/ipu_disp.c
+++ b/drivers/mxc/ipu3/ipu_disp.c
@@ -1108,7 +1108,7 @@ int32_t ipu_init_sync_panel(struct ipu_soc *ipu, int disp, uint32_t pixel_clk,
 		dev_err(ipu->dev, "clk_get di1 failed");
 		return PTR_ERR(ldb_di1_clk);
 	}
-	if (!strcmp(__clk_get_name(di_parent), __clk_get_name(ldb_di1_clk)) ||
+	if (!strcmp(__clk_get_name(di_parent), __clk_get_name(ldb_di0_clk)) ||
 		!strcmp(__clk_get_name(di_parent), __clk_get_name(ldb_di1_clk))) {
 		/* if di clk parent is tve/ldb, then keep it;*/
 		dev_dbg(ipu->dev, "use special clk parent\n");
-- 
1.7.9.5

